src/rules/orderedImportsRule.ts

```diff
@@ -127,64 +170,105 @@ export class Rule extends Lint.Rules.AbstractRule {
 type Transform = (x: string) => string;
 const TRANSFORMS = new Map<string, Transform>([
     ["any", () => ""],
-    ["case-insensitive", (x) => x.toLowerCase()],
+    ["case-insensitive", x => x.toLowerCase()],
     ["lowercase-first", flipCase],
-    ["lowercase-last", (x) => x],
-    ["full", (x) => x],
-    ["basename", (x) => {
-        if (!ts.isExternalModuleNameRelative(x)) {
-            return x;
-        }
+    ["lowercase-last", x => x],
+    ["full", x => x],
+    [
+        "basename",
+        x => {
+            if (!ts.isExternalModuleNameRelative(x)) {
+                return x;
+            }
 
-        const splitIndex = x.lastIndexOf("/");
-        if (splitIndex === -1) {
-            return x;
+            const splitIndex = x.lastIndexOf("/");
+            if (splitIndex === -1) {
+                return x;
+            }
+            return x.substr(splitIndex + 1);
         }
-        return x.substr(splitIndex + 1);
-    }],
+    ]
 ]);
 
 enum ImportType {
     LIBRARY_IMPORT = 1,
     PARENT_DIRECTORY_IMPORT = 2, // starts with "../"
-    CURRENT_DIRECTORY_IMPORT = 3, // starts with "./"
+    CURRENT_DIRECTORY_IMPORT = 3 // starts with "./"
 }
 
 interface Options {
     groupedImports: boolean;
     importSourcesOrderTransform: Transform;
-    moduleSourcePath: Transform;
+    moduleSourceTransform: Transform;
     namedImportsOrderTransform: Transform;
+    groups: GroupOption[];
+}
+
+interface GroupOption {
+    name: string;
+    match: RegExp | null;
+    order: number;
 }
 
 interface JsonOptions {
     "grouped-imports"?: boolean;
     "import-sources-order"?: string;
     "named-imports-order"?: string;
     "module-source-path"?: string;
+    groups?: JsonGroupOption[];
+}
+
+interface JsonGroupOption {
+    name?: string;
+    match: string | null;
+    order: number;
 }
 
 function parseOptions(ruleArguments: any[]): Options {
     const optionSet = (ruleArguments as JsonOptions[])[0];
+
+    const defaultGroups: JsonGroupOption[] = [
+        { name: "parent directories", match: "^[.][.]", order: 20 },
+        { name: "current directory", match: "^[.]", order: 30 },
+        { name: "libraries", match: null, order: 10 }
+    ];
+
     const {
         "grouped-imports": isGrouped = false,
         "import-sources-order": sources = "case-insensitive",
         "named-imports-order": named = "case-insensitive",
         "module-source-path": path = "full",
-    } = optionSet === undefined ? {} : optionSet;
+        groups: groups = defaultGroups
```

Comment:
no need for `: groups` since the name is the same. please remove

Reply:
done.  good catch.